### ISSUE TITLE
URGENT: GitHub is aggressively pruning caches

### ISSUE THREAD

**Issue by geekosaur** [haskell/cabal#10669]
We used to be allowed to use 300GB of cache, but starting this evening (quite possibly at 00:00 UTC) it's down to 25GB. I found this because it breaks the new CI setup completely, but even in the original `validate.yml` `dogfooding` relies on the cache to provide the `cabal` executable to be tested.

It's possible that jettisoning the store will get the cache down to a size GitHub will continue to cache, but worst case we may need to switch to using artifacts and just give up on cache entirely.

**Comment by geekosaur** [comment]
Okay, #10663 got in anyway, but that may end up on load somewhere. `dogfooding` should still get fixed to use the artifact.

**Comment by geekosaur** [comment]
And, I looked it over again and I managed to get the comment backwards: it says it _could_ use the cache, but there were reasons not to. It uses the artifact already, and it rebuilding more than necessary because the cabal cache is missing so it's taking longer than usual. So it's just the CI rewrite that needs to be modified.

My first attempt will probably keep the cache but remove the cabal store from it, since that's what bloats it and it'll never fit in GitHub's new cache limit.
